paddle/fluid/operators/slice_op.cc

```diff
@@ -155,8 +155,10 @@ class SliceOp : public framework::OperatorWithKernel {
 #ifdef PADDLE_WITH_MKLDNN
       auto input_data_type =
           framework::OperatorWithKernel::IndicateVarDataType(ctx, "Input");
-
-      if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
+      auto vec_dims = phi::vectorize(in_tensor.dims());
+      bool not_all_zero_dim = std::all_of(
+          vec_dims.cbegin(), vec_dims.cend(), [](int64_t i) { return i != 0; });
+      if (!not_all_zero_dims && this->CanMKLDNNBeUsed(ctx, input_data_type)) {
```

Comment:
You can remove negation now.
```suggestion
      bool all_zero_dim = std::all_of(
          vec_dims.cbegin(), vec_dims.cend(), [](int64_t i) { return i == 0; });
      if (all_zero_dims && this->CanMKLDNNBeUsed(ctx, input_data_type)) {
```